### PR TITLE
New 7 post review id comments

### PR DESCRIPTION
**Pull request by sarah-janewilson**

### CODE REVIEW

File: db/models/comments.model.js [hunk 1]
```diff
@@ -7,9 +7,25 @@ exports.fetchCommentsByReviewId = (fetchedComments) => {
...
+    .query(
+      `INSERT INTO comments (review_id, author, body)
+      VALUES ($1, $2, $3)
+      RETURNING*;`,
+      [reviewId, username, body]
+    )
+    .then((result) => {
+      const comment = { ...result.rows[0] };
+      delete comment.author;
+      comment.username = result.rows[0].author;
```
[the remark this answers is not in the document]
Comment (sarah-janewilson) [reply db/models/comments.model.js:28]: Ah yes, I read the ticket badly!

File: db/controllers/comments.controller.js [hunk 2]
```diff
@@ -8,3 +9,24 @@ exports.getAllCommentsByReviewId = (request, response, next) => {
...
 };
+
+exports.postNewComment = (request, response, next) => {
+  const { username, body } = request.body;
+  const reviewId = request.params.review_id;
+  if (!username || !body) {
+    return response
+      .status(400)
+      .send({ message: "Bad Request - Missing Required Fields" });
+  } else if (isNaN(reviewId)) {
```
Comment (two-triangles) [inline db/controllers/comments.controller.js:20]: postgresql will throw an error if the reviewId is not a number - I have a feeling you have some error handling middleware that might already be handling this.

File: db/models/comments.model.js [hunk 3]
```diff
@@ -7,9 +8,25 @@ exports.fetchCommentsByReviewId = (fetchedComments) => {
...
+    .query(
+      `INSERT INTO comments (review_id, author, body)
+      VALUES ($1, $2, $3)
+      RETURNING*;`,
+      [reviewId, username, body]
+    )
+    .then((response) => {
+      return response.rows[0];
+    })
+    .catch((err) => {
```
Comment (two-triangles) [inline db/models/comments.model.js:29]: You already have a catch block in your controller that will catch any errors from your model. Remember that essentially you model is returning a Promise chain that we are accessing the fulfilment of in the controller. If whatever the model is returning were to reject, the catch block in the controller would catch this.

Comment (sarah-janewilson) [reply db/models/comments.model.js]: I was just missing a 'next' somewhere so had added in all these extra bits to pass the tests!!

File: app.js [hunk 4]
```diff
@@ -32,6 +31,8 @@ app.use((err, request, response, next) => {
     response.status(err.status).send({ message: err.message });
   } else if (err.code === "22P02") {
     response.status(400).send({ message: "Invalid Review ID" });
+  } else if (err.code === "23503") {
+    response.status(404).send({ message: "Review Not Found" });
   } else response.status(500).send({ message: "Internal Server Error" });
 });
```
Comment (cjplatten) [inline app.js:37]: You can start to think about extracting these into their own file if you want!

**Review by cjplatten — COMMENTED** [review]
Looks good!

File: app.js (hunk 4 above)
Comment (sarah-janewilson) [reply app.js:37]: Yes will do!

**Pull request merged**